frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -110,12 +110,18 @@ export const eventListener: TelemetryEventListener = async (
   eventType: string,
   properties?: any,
 ) => {
-  if (TELEMETRY_DEBUG) {
+  if (TELEMETRY_DISABLED || !apiKey) {
```

Comment:
There might also be some confusion about this conditional block because there is no case where it will evaluate to true if TELEMETRY_DISABLED === true since this entire extension will not be resolved in that case.